setup.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/sh
+cd `dirname $0`
+
+# Create a virtual environment to run our code
+VENV_NAME="venv"
+PYTHON="$VENV_NAME/bin/python"
+ENV_ERROR="This module requires Python, pip, and virtualenv to be installed."
+
+if ! python3 -m venv $VENV_NAME; then
+    echo "Failed to create virtualenv." >&2
+    # If the virtualenv creation fails, we try to install python3-venv
+    if command -v apt-get >/dev/null; then
```

Comment:
I would flip this around:

```bash
if ! command -v apt-get >/dev/null; then
    echo "$ENV_ERROR" >&2
    exit 1
fi
echo "Detected Debian/Ubuntu..."
...
```
This keeps lines 13-35 less indented than they would otherwise be, and makes it easier on the reader because they don't need to keep all this extra state in their head when they get to line 37.